extensions/spatial.sigmf-ext.md

```diff
@@ -0,0 +1,385 @@
+
+Multichannel datasets can be stored in SigMF Collections or as multichannel
+interleaved data depending on the application. Collections are RECOMMENDED due
+to the easier processing and better application support.
+
+The `spatial` extension makes use of cartesian coordinates to define array
+geometry, and spherical coordinates for reporting bearings. This, coupled with
+the various methods for defining boresights and bearing references can become
+complicated rapidly so the union of these two cordinate reference systems (CRS)
+is illustrated in Figure 1 below. The ISO 80000-2:2019 compliant cartesian and
+spherical systems have a specific relationship with the reported azimuth and
+elevation, which use the conventional geospatial definitions (degrees east of
+north, and degrees above horizon respectively).
+
+![SigMF Spatial Coordinate Reference System](spatial_crs.png)
+
+**Figure 1 - SigMF Spatial Extension Coordinate Reference Systems**
+
+As shown in Figure 1, the boresight of the aperture is defined as being in the
+direction of the positive X-axis. Based on this, as an example, a horizontal
+uniform linear array aperture would be defined along the Y-axis (see examples
+section below).
+
+## 0 Datatypes
+
+This extension defines the following datatypes:
+
+|name|long-form name|description|
+|----|--------------|-----------|
+|bearing|signal direction bearing|JSON [bearing](spatial.sigmf-ext.md#01-the-bearing-object) object containing a quantitative representation of a direction with optional error fields.|
+|cartesian_point|cartesian position|JSON [cartesian_point](spatial.sigmf-ext.md#02-the-cartesian-point-object) containing a cartesian coordinate point triplet.|
+
+### 0.1 The `bearing` Object
+
+A `bearing` object is used to describe relative one or two dimensional signal
+directions. The units are always in degrees; while it is legal for this field to
+have any value it is RECOMMENDED that it be between 0 and 360 or +/- 180.
+
+|name|required|type|units|description|
+|----|--------|----|-----|-----------|
+|`azimuth`|false|float|degrees|Azimuth component of the direction in degrees increasing clockwise.|
+|`elevation`|false|float|degrees|Elevation component of the direction in degrees above horizon.|
+|`range`|false|float|meters|Line-of-sight slant range to emitter, if known, in meters.|
+|`az_error`|false|float|degrees|Error or uncertainty in the azimuth component.|
+|`el_error`|false|float|degrees|Error or uncertainty in the elevation component.|
+|`range_error`|false|float|meters|Error or uncertainty in the range component.|
+
+The `az_error`, `el_error`, and `range_error` field units are degrees, but the
+exact meaning of 'error' in this context is not explicitly defined. Applications
+SHOULD specify their specific meaning, and in general this should be interpreted
+as an uncertainty range. The error fields SHOULD NOT be included if the
+corresponding estimate fields are not present.
+
+```json
```

Comment:
Recommend adding a note about this indicating it's an example describing some kind of bearing